chain/network/src/peer_manager.rs

```diff
@@ -355,6 +397,9 @@ impl PeerManagerActor {
         remove_from_peer_store: bool,
     ) {
         debug!(target: "network", "Unregister peer: {:?} {:?}", peer_id, peer_type);
+
+        self.routing_table.remove_peer(&peer_id);
```

Comment:
This should only be called inside `remove_active_peer` but not here. If `remove_from_peer_store` is false, and this function is called, data from an active peer might get removed. Read more [here](https://github.com/near/nearcore/blob/master/chain/network/src/peer.rs#L682-L688).